src/state/opamRepositoryState.ml

```diff
@@ -147,9 +147,13 @@ let get_repo_root rt repo =
 let load lock_kind gt =
   log "LOAD-REPOSITORY-STATE %@ %a" (slog OpamFilename.Dir.to_string) gt.root;
   let lock = OpamFilename.flock lock_kind (OpamPath.repos_lock gt.root) in
-  let repos_map =
-    OpamFile.Repos_config.safe_read (OpamPath.repos_config gt.root)
-  in
+  let repos_map = OpamStateConfig.Repos.safe_read ~lock_kind gt in
+  if OpamStateConfig.is_readonly_opamroot gt then
```

Comment:
hmm that confused me, maybe `is_readonly_opamroot` could be better named. Stg like `is_newer_than_self` ?